### DISCUSSION TITLE
Deployment view like Structurizr

### DISCUSSION THREAD

**Discussion by Kiiv** [likec4/likec4#1245] [Ideas]
Hi !

Are you planning to implement something similar to Deployment View in Structurizr ?
https://docs.structurizr.com/ui/diagrams/deployment-view

Or have I missed a feature that already allowing to do it ?
Groups does not seems to fit the need as stated here : https://github.com/likec4/likec4/discussions/1175

Regards,
Clément.

**Comment by Cyclonit** [comment]
@hubertmis posted a proposal similar to Structurizr's deployment views here: https://github.com/likec4/likec4/discussions/533#discussioncomment-11221969

I agree that this would be a great feature and if I understood @davydkov correctly, he wanted to work on a proposal for this. It'd be good if we could collect some requirements, such that they can be incorporated from the beginning. Given that breaking changes in the DSL are bad, the clearer the picture is from the beginning the better.

**Comment by Kiiv** [comment]
Thanks for your quick reply !

Comment you pointed me to seems a really good start.
What come to my mind at the moment :
- deployment nodes could be imbricated : a node "virtual machine" could contain one or more nodes "Application Serveur" containing one or more container instances
- a container could be instanciated on one or more deployment node
- it could be possible to restrict the relationships between container to a "cluster"
For example : 
container A have a relationship to container B 
container A is deployed on node "site1-A" and  on node "site2-A"
container B is deployed on node "site1-B" and  on node "site2-B"
instance of container A on site1-A must not have a relationship with instance of container B on site2-B but only with instance B on site1-B
- it should be really great to manage proxies, load balancer etc in an easy way
In container view, A has a direct relationship to B. But in deployment environment there is a reverse proxy between A and B

**Reply by davydkov** [reply]
Hey @Kiiv 

What do you think about #1250? (work in progress)

**Comment by davydkov** [comment]
Move discussion to #1269
